image_tools/src/cam2image.cpp

```diff
@@ -118,55 +96,151 @@ class Cam2Image : public rclcpp::Node
+  }
+
+  /// Publish camera, or burger, image.
+  IMAGE_TOOLS_LOCAL
+  void timerCallback()
+  {
     cv::Mat frame;
-    cv::Mat flipped_frame;
-
-    size_t i = 1;
-    // Our main event loop will spin until the user presses CTRL-C to exit.
-    while (rclcpp::ok()) {
-      // Initialize a shared pointer to an Image message.
-      auto msg = std::make_unique<sensor_msgs::msg::Image>();
-      msg->is_bigendian = false;
-      // Get the frame from the video capture.
-      if (burger_mode_) {
-        frame = burger_cap.render_burger(width_, height_);
-      } else {
-        cap >> frame;
-      }
-      // Check if the frame was grabbed correctly
-      if (!frame.empty()) {
-        // Convert to a ROS image
-        if (!is_flipped) {
-          convert_frame_to_message(frame, i, *msg);
-        } else {
-          // Flip the frame if needed
-          cv::flip(frame, flipped_frame, 1);
-          convert_frame_to_message(flipped_frame, i, *msg);
-        }
-        if (show_camera_) {
-          cv::Mat cvframe = frame;
-          // Show the image in a window called "cam2image".
-          cv::imshow("cam2image", cvframe);
-          // Draw the image to the screen and wait 1 millisecond.
-          cv::waitKey(1);
-        }
-        // Publish the image message and increment the frame_id.
-        RCLCPP_INFO(node_logger, "Publishing image #%zd", i);
-        pub_->publish(std::move(msg));
-        ++i;
-      }
-      // Do some work in rclcpp and wait for more to come in.
-      loop_rate.sleep();
+
+    // Initialize a shared pointer to an Image message.
+    auto msg = std::make_unique<sensor_msgs::msg::Image>();
+    msg->is_bigendian = false;
+
+    // Get the frame from the video capture.
+    if (burger_mode_) {
+      frame = burger_cap.render_burger(width_, height_);
+    } else {
+      cap >> frame;
+    }
+
+    // If no frame was grabbed, return early
+    if (frame.empty()) {
+      return;
     }
+
+    // Conditionally flip the image
+    if (is_flipped_) {
+      cv::flip(frame, frame, 1);
+    }
+
+    // Convert to a ROS image
+    convert_frame_to_message(frame, publish_number_, *msg);
+
+    // Conditionally show image
+    if (show_camera_) {
+      cv::Mat cvframe = frame;
+      // Show the image in a window called "cam2image".
+      cv::imshow("cam2image", cvframe);
+      // Draw the image to the screen and wait 1 millisecond.
+      cv::waitKey(1);
+    }
+
+    // Publish the image message and increment the frame_id.
+    RCLCPP_INFO(get_logger(), "Publishing image #%zd", publish_number_++);
+    pub_->publish(std::move(msg));
+  }
+
+  IMAGE_TOOLS_LOCAL
+  bool help(const std::vector<std::string> args)
+  {
+    if (std::find(args.begin(), args.end(), "--help") != args.end() ||
+      std::find(args.begin(), args.end(), "-h") != args.end())
+    {
+      std::stringstream ss;
+      ss << "Usage: cam2image [-h] [--ros-args [-p param:=value] ...]" << std::endl;
```

Comment:
Nitpick: each `std::endl` also flushes the stream which doesn't seem necessary on each line.